### ISSUE TITLE
Uploaded Folders result loosing Folder Structure and renamed Files

### ISSUE THREAD

**Issue by capekasa** [mifi/ezshare#76]
When downloading a folder ezshare zips the contents automatically on the fly which is great. 
But this doesn't happen when uploading from a client (e.g. Firefox/Chrome on Mac OS to a Windows PC running ezshare). Instead the folder structure will be flattened, all files renamed thus rendering everything pretty much useless if filenames and folders matter. They do!

e.g.
"Foldername/IMG_1234.JPG" results in the file "Foldername!IMG_1234.JPG"

Unclear if that

**Comment by mifi** [comment]
I think we need to implement support for detecting and recreating a deep folder structure on the server. did you drag-drop a folder or how did you produce this problem?

**Comment by capekasa** [comment]
Yes, I did drag&drop the folder. Is ezshare zipping at all during uploads?

**Comment by mifi** [comment]
> Is ezshare zipping at all during uploads?

no

**Comment by aalkz** [comment]
Hello! I just encountered this exact issue. I transferred a mix of files and folders by drag and dropping them into the web interface and everything copied over correctly but the folder structure didn't carry over, it just put the folder name at the beginning of each file that was nested just like @capekasa described.

Source:
<img width="629" height="169" alt="Image" src="https://github.com/user-attachments/assets/dc8aace2-4c69-4872-aa1d-abb7e1bf596f" />

Destination:
<img width="685" height="197" alt="Image" src="https://github.com/user-attachments/assets/3f26217c-bad1-450b-b66f-f0d4a0309e9d" />

<img width="349" height="233" alt="Image" src="https://github.com/user-attachments/assets/b836be55-c873-4273-80c5-829c4db5776a" />

It also happens if I just drag and drop the folder by itself so I don't think it's because I mixed files and folders.
